### DISCUSSION TITLE
Roborock S5Max Error 500

### DISCUSSION THREAD

**Discussion by SPagesDe** [Hypfer/Valetudo#1261] [Q&A / Support] [answered]
Hello Hypfer,
I have rooted my Roborock S5max and installed Valetudo. Everything worked fine for the first few days. But now almost all requests run into an "Internal Server Error 500".

The request ".../api/v2/robot/" still returns a response "{"manufacturer": "Roborock", "modelName": "S5 Max", "implementation": "RoborockS5MaxValetudoRobot"}".

The request ".../api/v2/robot/state" already leads into an error 500.

Vacuuming in remote mode works fine too. A full cleaning directly at the robot started also.

Do you have an idea in which direction I could look for the error? Can I try any commands through the console (I have SSH access)?

Thanks for your great piece of software.
SPages

<details>
<summary>
[DE]
</summary>

Hallo Hypfer,
ich habe meinen Roborock S5max gerootet und Valetudo installiert. In den ersten Tagen hat auch alles gut funktioniert. Nun laufen aber fast alle anfragen in einen "Internal Server Error 500".

Die Anfrage ".../api/v2/robot/" liefert noch eine Antwort "{"manufacturer":"Roborock","modelName":"S5 Max","implementation":"RoborockS5MaxValetudoRobot"}"

Die Anfrage ".../api/v2/robot/state" führt schon in einen Fehler 500.

Staubsaugen im Fernbedinungsmodus funktioniert auch problemlos. Eine voll Reinigung direkt am Roboter gestartet auch.

Hast du eine Idee in welche Richtung ich nach dem Fehler suchen könnte? Kann man irgendwelche Befehle über die Konsole versuchen (habe SSH Zugang)?

Schon einmal vielen Dank für deine tolles Stück Software.
SPages
</details>

**Comment by Hypfer** [comment]
Anything in the log?

**Comment by SPagesDe** [comment]
Yes. Again and again 

[2021-12-05T12:22:02.397Z] [INFO] Miio error response {
  id: 7,
  error: { code: -10002, message: 'Access denied' },
  exe_time: 22
}



<details>
<summary>The log fully ...</summary>

```
[2021-12-05T12:21:16.977Z] [INFO] Set Logfile to /tmp/valetudo.log
[2021-12-05T12:21:16.986Z] [INFO] Autodetected RoborockS5MaxValetudoRobot
[2021-12-05T12:21:17.098Z] [INFO] Starting Valetudo 2021.11.1
[2021-12-05T12:21:17.100Z] [INFO] Commit ID: 5cf5e5475476e3b930421f14eb95f343f52b292a
[2021-12-05T12:21:17.101Z] [INFO] Configuration file: /mnt/data/valetudo_config.json
[2021-12-05T12:21:17.102Z] [INFO] Logfile: /tmp/valetudo.log
[2021-12-05T12:21:17.104Z] [INFO] Robot: Roborock S5 Max (RoborockS5MaxValetudoRobot)
[2021-12-05T12:21:17.104Z] [INFO] JS Runtime Version: v16.10.0
[2021-12-05T12:21:17.105Z] [INFO] Arch: arm
[2021-12-05T12:21:17.107Z] [INFO] Max Heap Size: 34 MiB
[2021-12-05T12:21:17.108Z] [INFO] Node Flags: --expose-gc --max-heap-size=34 --optimize-for-size --lite-mode --no-regexp-tier-up --no-expose-wasm
[2021-12-05T12:21:17.121Z] [INFO] Autogenerated System ID: SlimBoldStingray
[2021-12-05T12:21:17.137Z] [INFO] DeviceId 
[2021-12-05T12:21:17.137Z] [INFO] IP 127.0.0.1
[2021-12-05T12:21:17.138Z] [INFO] CloudSecret 
[2021-12-05T12:21:17.139Z] [INFO] LocalSecret 
[2021-12-05T12:21:17.144Z] [INFO] Firmware Version: 1530
[2021-12-05T12:21:17.148Z] [INFO] Current rootfs: system_a (nande)
[2021-12-05T12:21:35.289Z] [INFO] Valetudo can be reached via: valetudo-slimboldstingray.local
[2021-12-05T12:21:35.326Z] [INFO] Setting /proc/self/oom_score_adj to 666. Previous value: 0
[2021-12-05T12:21:35.342Z] [INFO] Dummycloud is spoofing 127.0.0.1:8053 on 127.0.0.1:8053
[2021-12-05T12:21:35.353Z] [INFO] Webserver running on port 80
[2021-12-05T12:21:35.366Z] [INFO] SSDP/UPnP advertisement started
[2021-12-05T12:21:35.381Z] [INFO] Map Upload Server running on port 8079
[2021-12-05T12:21:35.407Z] [INFO] Bonjour service "Valetudo S5 Max Web" with type "http" started
[2021-12-05T12:21:35.407Z] [INFO] Bonjour service "Valetudo S5 Max" with type "valetudo" started
[2021-12-05T12:21:35.417Z] [WARN] Error during time sync: Error: Timeout waiting for NTP response from pool.ntp.org:123
    at Object.getNetworkTime (/snapshot/Valetudo/node_modules/@destinationstransfers/ntp/index.js:64:11)
    at async NTPClient.pollTime (/snapshot/Valetudo/backend/lib/NTPClient.js:56:36)
[2021-12-05T12:21:36.219Z] [INFO] Cloud connected
[2021-12-05T12:21:36.361Z] [INFO] Unknown cloud message received: {"id":0,"result":"unknown_method"}
[2021-12-05T12:21:41.755Z] [INFO] Miio error response {
  id: 3,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:21:49.964Z] [INFO] Miio error response {
  id: 4,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:21:49.968Z] [WARN] Error while calling BasicControlCapability Action start { code: -10002, message: 'access denied' }
[2021-12-05T12:21:51.724Z] [INFO] Miio error response {
  id: 5,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:21:51.728Z] [WARN] Error while calling BasicControlCapability Action home { code: -10002, message: 'access denied' }
[2021-12-05T12:22:02.397Z] [INFO] Miio error response {
  id: 7,
  error: { code: -10002, message: 'access denied' },
  exe_time: 22
}
[2021-12-05T12:22:07.853Z] [INFO] Miio error response {
  id: 9,
  error: { code: -10002, message: 'access denied' },
  exe_time: 17
}
[2021-12-05T12:22:09.463Z] [INFO] Miio error response {
  id: 10,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:10.488Z] [INFO] Miio error response {
  id: 11,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:12.515Z] [INFO] Miio error response {
  id: 12,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:16.547Z] [INFO] Miio error response {
  id: 13,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:22.755Z] [INFO] Miio error response { id: 14, error: { code: -10002, message: 'access denied' } }
[2021-12-05T12:22:33.512Z] [INFO] Miio error response {
  id: 15,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:33.524Z] [INFO] Miio error response {
  id: 16,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:34.536Z] [INFO] Miio error response {
  id: 17,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:34.557Z] [INFO] Miio error response {
  id: 18,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:34.005Z] [INFO] Successfully set the robot time via NTP to 2021-12-05T12:22:34.895Z
[2021-12-05T12:22:34.801Z] [INFO] Miio error response {
  id: 19,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:34.809Z] [INFO] Miio error response {
  id: 20,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:50.793Z] [INFO] Miio error response {
  id: 21,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:50.804Z] [INFO] Miio error response {
  id: 22,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:51.817Z] [INFO] Miio error response {
  id: 23,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:51.856Z] [INFO] Miio error response {
  id: 24,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:53.908Z] [INFO] Miio error response {
  id: 25,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:53.920Z] [INFO] Miio error response {
  id: 26,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:57.944Z] [INFO] Miio error response {
  id: 27,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:22:57.955Z] [INFO] Miio error response {
  id: 28,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:06.977Z] [INFO] Miio error response {
  id: 29,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:07.014Z] [INFO] Miio error response {
  id: 30,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:12.488Z] [INFO] Miio error response {
  id: 31,
  error: { code: -10002, message: 'access denied' },
  exe_time: 2
}
[2021-12-05T12:23:12.498Z] [INFO] Miio error response {
  id: 32,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:14.518Z] [INFO] Miio error response {
  id: 33,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:14.538Z] [INFO] Miio error response {
  id: 34,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:18.544Z] [INFO] Miio error response {
  id: 35,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:18.565Z] [INFO] Miio error response {
  id: 36,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:34.974Z] [INFO] Miio error response {
  id: 37,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:35.012Z] [INFO] Miio error response {
  id: 38,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:36.007Z] [INFO] Miio error response {
  id: 39,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:36.034Z] [INFO] Miio error response {
  id: 40,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:38.037Z] [INFO] Miio error response {
  id: 41,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}
[2021-12-05T12:23:38.062Z] [INFO] Miio error response {
  id: 42,
  error: { code: -10002, message: 'access denied' },
  exe_time: 1
}

```

</details>

**Reply by Hypfer** [reply]
That's the region lock feature being very unhappy.

Did you by chance edit the `/mnt/default/roborock.conf`?
Because if yes, I hope you have a backup of that.

See also: https://github.com/Hypfer/Valetudo/discussions/924

**Comment by SPagesDe** [comment]
Yes ... I might have changed something ... f*** ... but all backups have the same content ...

language=us
name=custom_A.03.0070_CE
bom=A.03.0070
location=de
wifiplan=
timezone=Europe/Berlin
logserver=awsde0.fds.api.xiaomi.com
featureset=1
cpuid=6c1...

I suppose I do not find a backup am owner of e-waste?

**Reply by Hypfer** [reply]
Same content as in same text or same bytes?

**Comment by SPagesDe** [comment]
The same bytes ... in the backups it says "language=us", but I have imported it without success


and also a try after has brought nothing  
https://github.com/Hypfer/Valetudo/discussions/924

... even if I create a \r\n at the end I get the same not good result.

CPU ID I just compared ... are both the same (/mnt/reserve/hwinfo)

**Comment by SPagesDe** [comment]
Is there a way to give a rooted S5max a factory reset - so that it forgets all changes again? 

Rooting again would not be a big effort

**Reply by Hypfer** [reply]
Factory resetting doesn't restore the /mnt/default and /mnt/reserve partiton.

You will need the backup created during the rooting process

**Reply by SPagesDe** [reply]
I did, but I have taken various files from it via Windows and restored via SSH ... without success

How to restore the backups? Certainly via UART to the vacuum cleaner and then?

**Reply by SPagesDe** [reply]
Okay I think I found what I was looking for

dd if=/mnt/data/backups/nandb.dd of=/dev/nandb

nandb can be unmounted ... nandk (/mnt/reserve) not. But it makes sense to unmount all three partitions or? So nandA, B and K?

**Comment by SPagesDe** [comment]
I have restored the backups. In the meantime, Valetudo had also displayed my map again. However, this disappeared again after a short time and the errors are back.

**Reply by Hypfer** [reply]
Please open a ticket at https://vaers.dontvacuum.me/

**Reply by SPagesDe** [reply]
There is already a ticket running for this.

What I am a bit surprised that when I tried through the "Swagger UI" yesterday very many commands worked. Water quantity, vacuum cleaner performance, clean area, go to position work normally with a PUT call via Curl (I had built a small PHP script for this). 

Only areas like "BasicControlCapability" (home, stop, pause) and ".../robot/state" lead to the error.

But probably completely different "places" are addressed for it in the background.

Thanks to you already Hypfer for your help. I will download the sourcecode of the first version of Valetudo this weekend and have a look at it. Because I can program a little bit and maybe I will understand better how everything works in the background. Even if the architecture of "the robot" has probably changed between V1 and S5max with V3 (?).

**Reply by Hypfer** [reply]
You will not find anything in Valetudo.
This is exclusively caused by the region lock feature of the firmware.

**Reply by SPagesDe** [reply]
I already understood that Valetudo "only" accepts requests and passes them on to an interface of the vacuum cleaner robot.

But I am interested in exactly this place and how it works with the token, etc. to better understand the technology behind it.

**Comment by SPagesDe** [comment]
Dieser Text auf deutsch unter [DE] (This text in german under [DE])

**[EN]**
As Hypfer already wrote, this error is probably really a problem with the "country lock". Currently I have everything that I thought was likely to CN

By setting the following parameters to CN (for China) you get access to your vacuum cleaner again. Please make a backup first (Linux function dd -> partition list at https://github.com/dgiese/dustcloud/wiki/Partition-Information)

Change entries best from the beginning of the line and then delete superfluous characters with the remove key (<- reason: different line breaks are used) 

**Structure of the following table**
No. - has effect on geoblocking? | path -> file => change or (alt ...)

________________
**YES**
________________

1.**YES** | /mnt/data/miio -> device.country => de -> cn

2.**_(YES)_** | /mnt/data/rockrobo -> country_info
	==> This entry is automatically created from /mnt/data/miio/device.country at boot time.
	

________________
NO
________________

Where else is something with "Country, Timezone", but which (currently) does not lead to geoblocking.


3. NO | /mnt/data/miio -> wifi.conf => de -> cn

+++++++> get read and write permissions for /mnt/default - mount -o remount,rw /mnt/default

4. NO | /mnt/default/roborock.conf => language=cn (old us)
							  location=cn (old de)
							  timezone=Asia/Shanghai (old Europe/Berlin)
							  
	 ==> info about "roborock.conf.sign"
	 [...]The problem is the signature. If you create a firmware via https://builder.dontvacuum.me it has the signature check "turned off"[...]
	 
	 -> so you don't have to worry about the signature in case of a firmware https://builder.dontvacuum.me (currently).
	 (Dennis Giese writes about this at https://www.roboter-forum.com/index.php?thread/45980-region-geo-blocking-entfernen-t6-t4/&pageNo=1 - it's the same with S5Max)
	 
							
5. NO | /mnt/default/timezone => Europe/Berlin -> Asia/Shanghai				

(Created on 29.12.2021)

<details>
<summary>
**[DE]**
</summary>

Es handelt sich bei diesem Fehler wie Hypfer schon schrieb wohl wirklich um ein Problem mit der "Ländersperre". Aktuell habe ich mal alles was ich für wahrscheinlich gehalten habe auf CN

Durch setzten folgender Parameter auf CN (für China) bekommt ihr wieder zugriff auf euren Staubsauger. Bitte macht vorher auf jeden Fall ein Backup (Linux Funktion dd -> Partitionsliste unter https://github.com/dgiese/dustcloud/wiki/Partition-Information)

Änderung von Einträgen am besten vom Zeilenanfang und dann überflüssige Zeichen per entfernen Taste löschen (<- Grund: es werden verschiedene Zeilenumbrüche verwendet) 

**Aufbau der folgenden Tabelle**
Nr. - hat Auswirkung auf Geoblocking? | Pfad -> Datei => änderung bzw. (alt ...)

________________
**JA**
________________

1. **JA** | /mnt/data/miio -> device.country => de -> cn

2. **_(JA)_** | /mnt/data/rockrobo -> country_info
	==> Dieser Eintrag wird automatisch aus /mnt/data/miio/device.country beim Booten erzeugt.
	

________________
NEIN
________________

Wo sonst noch etwas mit "Country, Timezone" steht, was aber (aktuell) nicht zu Geoblocking führt ist.


3. NEIN | /mnt/data/miio -> wifi.conf	=> de -> cn

+++++++> Lese- und Schreibrechte bekommen für /mnt/default - mount -o remount,rw /mnt/default

4. NEIN | /mnt/default/roborock.conf => language=cn (old us)
							  location=cn (old de)
							  timezone=Asia/Shanghai (old Europe/Berlin)
							  
	 ==> Info zur "roborock.conf.sign"
	 [...]Das Problem ist die Signatur. Wenn man eine Firmware über https://builder.dontvacuum.me erzeugt, hat diese die Signaturprüfung "ausgeschaltet".[...]
	 
	 -> über die Signatur müsst ihr euch also im Fall von einer Firmware https://builder.dontvacuum.me (aktuell) keine sorgen machen
	 (schreibt Dennis Giese dazu unter https://www.roboter-forum.com/index.php?thread/45980-region-geo-blocking-entfernen-t6-t4/&pageNo=1 - ist auch bei S5Max so)
	 
							
5. NEIN | /mnt/default/timezone 	=> 	Europe/Berlin -> Asia/Shanghai				

(Erstellt am 29.12.2021)

</details>

**Comment by SPagesDe** [comment]
So my Roborock is now reachable and controllable again as Chinese, but now it tries to reach "cn.ot.io.mi.com" and "cn.ott.io.mi.com" which are both not in the /etc/hosts-bind/hosts and therefore I get no map at least via the interface (I think). Changing the device.country to e.g. "de" leads back to the initial error. 

But the /etc/hosts-bind/hosts is "Read only". Is there any way to change this file?

Also, I guess it contacts the Alibaba Cloud (161.117.49.2, and 161.117.52.228)

Excerpt of the miio.log
```

[20211229 18:15:20] [I] set udp host to cn.ot.io.mi.com
[20211229 18:15:20] [I] set tcp host to cn.ott.io.mi.com
[20211229 18:15:23] [I] recv local cmd
[20211229 18:15:23] [I] local.query_status
[20211229 18:15:25] [I] cmd_internal_res_wifi_conf_status:1310 STATE: [STATE_TOKEN_DONE] -> [STATE_WIFI_STA_MODE]
[20211229 18:15:25] [I] parse_http_dns, http_dns fail, did: 391790066, dm: cn.ot.io.mi.com, host: 127.000.0.1
[20211229 18:15:25] [I] parse_http_dns, http_dns fail, did: 391790066, dm: cn.ott.io.mi.com, host: 127.000.0.1
[20211229 18:15:25] [I] getaddrinfo (cn.ot.io.mi.com:8053): succeed
[20211229 18:15:25] [I] Add UDP server into list, host ip: 161.117.49.2, port: 8053
[20211229 18:15:25] [I] Add UDP server into list, host ip: 161.117.52.228, port: 8053
```

**Reply by SPagesDe** [reply]
Okay with "mount --bind /mnt/reserve/hosts /etc/hosts" in the _root.sh you can change the hosts file. But when I add the CN domains here I get the initial error again.

But at http://ROBOROCK-IP/api/v2/robot/state/map I get map data as return value "{"__class": "ValetudoMap", "metaData":{"defaultMap":true, "version":2, ..." - I do not understand.

Log File:

```
Debug
>>> local: {"method":"get_status","params":{},"id":15}
2021-12-29T21:10:02.763Z
Debug
>>> local: {"method":"get_status","params":{},"id":16}
2021-12-29T21:10:02.788Z
Debug
<<< local: {"result":[{"msg_ver":2,"msg_seq":53,"state":5,"battery":100,"clean_time":30,"clean_area":575000,"error_code":0,"map_present":1,"in_cleaning":1,"in_returning":0,"in_fresh_state":0,"lab_status":1,"water_box_status":0,"fan_power":101,"dnd_enabled":1,"map_status":3,"is_locating":0,"lock_status":0,"water_box_mode":202,"water_box_carriage_status":0,"mop_forbidden_enable":0}],"id":15}
2021-12-29T21:10:02.802Z
Debug
<<< local: {"result":[{"msg_ver":2,"msg_seq":54,"state":5,"battery":100,"clean_time":30,"clean_area":575000,"error_code":0,"map_present":1,"in_cleaning":1,"in_returning":0,"in_fresh_state":0,"lab_status":1,"water_box_status":0,"fan_power":101,"dnd_enabled":1,"map_status":3,"is_locating":0,"lock_status":0,"water_box_mode":202,"water_box_carriage_status":0,"mop_forbidden_enable":0}],"id":16}
2021-12-29T21:10:02.869Z
Debug
>>> local: {"method":"get_status","params":{},"id":17}
2021-12-29T21:10:02.883Z
Debug
<<< local: {"result":[{"msg_ver":2,"msg_seq":55,"state":5,"battery":100,"clean_time":30,"clean_area":575000,"error_code":0,"map_present":1,"in_cleaning":1,"in_returning":0,"in_fresh_state":0,"lab_status":1,"water_box_status":0,"fan_power":101,"dnd_enabled":1,"map_status":3,"is_locating":0,"lock_status":0,"water_box_mode":202,"water_box_carriage_status":0,"mop_forbidden_enable":0}],"id":17}
```

**Comment by SPagesDe** [comment, marked as answer]
**[EN] I have won :D**

Okay, I was very close to the solution yesterday. Sometimes it helps to sleep one more night over it.

Yesterday I was so far that I could control my robot completely again, but it still didn't show me a map in Valetudo. But also this problem I could solve now :)

Continuation of the tutorial from the day before:
1. copy the hosts file from "/etc/hosts-bind" to "/mnt/reserve".

2. add the two China servers in the upper area (redirection of requests to "local cloud") - like this ...

```
#
# Add to /etc/hosts
#

127.000.0.1 cn.ot.io.mi.com
::1 cn.ot.io.mi.com
127.000.0.1 cn.ott.io.mi.com
::1 cn.ott.io.mi.com
```

3. add the following line to the "_root.sh" file

mount --bind /mnt/reserve/hosts /etc/hosts

With this, we mount our customized hosts file to the file system. Don't worry, if for some reason you don't want to do this anymore, just remove this mount point and the default values will be loaded again. The original hosts file will not be changed by this operation.

4. now set the timezone under "/mnt/data/rockrobo/" in the timezone to Asia/Shanghai.

5. reboot and we are done :)

<details>

<summary>
[DE] Ich habe gewonnen :D

</summary>

Okay, da war ich gestern schon sehr nahe dran an der Lösung. Mitunter hilft es halt noch einmal eine Nacht darüber zu schlafen.

Gestern war ich soweit das ich meinen Roboter wieder vollständig Steuern konnte, er mir jedoch in Valetudo weiterhin keine Karte angezeigt hat. Doch auch diese Problem konnte ich nun lösen :)

Fortsetzung der Anleitung vom Vortag:
1. Kopieren Sie die Datei hosts aus "/etc/hosts-bind" nach "/mnt/reserve"

2. Fügen sie im oberen Bereich die zwei China Server ein (Umleitung der Anfragen auf "lokale Cloud") - etwa so ...

```
#
# Add to /etc/hosts
#

127.000.0.1  cn.ot.io.mi.com
::1  cn.ot.io.mi.com
127.000.0.1  cn.ott.io.mi.com
::1  cn.ott.io.mi.com

```
3. Erweitern Sie die Datei "_root.sh" um folgende Zeile

mount --bind /mnt/reserve/hosts /etc/hosts

Damit binden wir unsere angepasste hosts Datei in das Filesystem ein. Keine Angst, wollte ihr dies aus irgendeinem Grund nicht mehr, nehmt ihr diesen mount Punkt einfach raus und es werden wieder die Default-Werte geladen. Die originale hosts Datei wird durch diesen Eingriff nicht verändert.

4. Nun noch die Zeitzone unter "/mnt/data/rockrobo/" in der timezone auf Asia/Shanghai stellen

5. Ein Reboot und wir sind fertig :)

</summary>
**In the end, of course, thanks again to you Hypfer that you have steered my thoughts in the right direction. This problem may be considered solved. Maybe this contribution will help someone in the future.**

**Reply by Hypfer** [reply]
Okay we can fix this in the image build process. Apparently, these domains are rather new?

https://github.com/dgiese/dustbuilder-features/blob/master/valetudo/deployment/etc/hosts-local

I've just updated the hosts file. This change will be applied whenever @dgiese updates the dustbuilder.
You won't need the bind mount anymore then.

So to recap, you've changed the timezone of the robot to something else than `Asia/Shanghai` which then triggered the region lock?

**Reply by SPagesDe** [reply]
My S5Max should not be a China device, because I bought it in Germany at a large online retailer with "A" in the name (The seller looks like "Roborock Germany").

What I have done that geoblocking has apparently struck I can not really say.

It seems that when the robot software detects "irregularities" it resets the robot to China.

These domains are as far as I can see now composed of the entry in "/mnt/data/miio/device.country" and followed by the domains.

So as seen here ...
[COUNTRY from device.country].ot.io.mi.com and [COUNTRY from device.country].ott.io.mi.com

I have to set the whole thing to China (cn), otherwise the geoblocking will lock me out.

**Reply by dgiese** [reply]
Change pushed to prod and staging dustbuilder.
That domain is fairly new, it was likely added in some Firmware update a while ago. Traditionally, the "cn" region was represented by an empty country file. Try to build a new firmware and see if it goes away. Ideally, your robot should behave like a global model. Otherwise we need to investigate further.
